tests/Integration/OptionsTest.php

```diff
@@ -0,0 +1,169 @@
+<?php
+/**
+ * Integration Tests: Plugin options
+ *
+ * @package Parsely\Tests
+ */
+
+declare(strict_types=1);
+
+namespace Parsely\Tests\Integration;
+
+use Parsely\Parsely;
+use SebastianBergmann\RecursionContext\InvalidArgumentException;
+use PHPUnit\Framework\ExpectationFailedException;
+
+/**
+ * Integration Tests for plugin options.
+ */
+final class OptionsTest extends TestCase {
+	/**
+	 * Internal variable.
+	 *
+	 * @var Parsely $parsely Holds the Parsely object.
+	 */
+	private static $parsely;
+
+	/**
+	 * Setup method called before each test.
+	 */
+	public function set_up(): void {
+		parent::set_up();
+
+		self::$parsely = new Parsely();
+	}
+
+	/**
+	 * Verifies that get_options() can handle corrupted or unset values.
+	 *
+	 * @since 3.0.0
+	 *
+	 * @covers \Parsely\Parsely::get_options
+	 * @uses \Parsely\Parsely::get_managed_credentials
+	 * @uses \Parsely\Parsely::set_default_track_as_values
+	 */
+	public function test_default_options_are_returned_when_options_are_corrupted_or_not_set(): void {
+		add_option( Parsely::OPTIONS_KEY, 'someinvalidvalue' );
+
+		$options = self::$parsely->get_options();
+		self::assertSame( self::$parsely->get_default_options(), $options );
+
+		delete_option( Parsely::OPTIONS_KEY );
+		self::assertSame( self::$parsely->get_default_options(), $options );
+	}
+
+	/**
+	 * Verifies that get_options() returns the correct "Track As" defaults.
+	 *
+	 * @since 3.9.0
+	 *
+	 * @covers Parsely\Parsely::get_options
+	 * @uses \Parsely\Parsely::get_managed_credentials
+	 * @uses \Parsely\Parsely::set_default_track_as_values
+	 */
+	public function test_get_options_returns_correct_track_as_defaults(): void {
+		$options = self::$parsely->get_options();
+		self::assertSame( array( 'post' ), $options['track_post_types'] );
+		self::assertSame( array( 'page' ), $options['track_page_types'] );
+	}
+
+	/**
+	 * Verifies that set_default_track_as_values() does not get called when
+	 * saved plugin options exist.
+	 *
+	 * @since 3.9.0
+	 *
+	 * @covers Parsely\Parsely::get_options
+	 * @uses \Parsely\Parsely::get_managed_credentials
+	 */
+	public function test_set_default_track_as_values_should_not_be_called_when_saved_options_exist(): void {
+		$options                     = self::$parsely->get_options();
+		$options['track_post_types'] = array( 'new_post_type' );
+		$options['track_page_types'] = array( 'new_page_type' );
+
+		add_option( Parsely::OPTIONS_KEY, $options );
+		$saved_options = self::$parsely->get_options();
+
+		self::assertSame( array( 'new_post_type' ), $saved_options['track_post_types'] );
+		self::assertSame( array( 'new_page_type' ), $saved_options['track_page_types'] );
+
+		delete_option( Parsely::OPTIONS_KEY );
```

Comment:
This cleanup shouldn't be necessary since the testcase class attempts to reset the state after each test.

Reply:
Removed and verified that tests work without issues.